src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_CallerArgumentExpression.vb

```diff
@@ -0,0 +1,1189 @@
+            compilation.VerifyDiagnostics()
+        End Sub
+
+        <ConditionalFact(GetType(CoreClrOnly))>
+        Public Sub TestSetter1()
+            Dim source As String = "
+Imports System
+Imports System.Runtime.CompilerServices
+
+Class Program
+    Public Property P As String
+        Get
+            Return ""Return getter""
+        End Get
+        Set(<CallerArgumentExpression("""")> value As String)
+            Console.WriteLine(value)
+        End Set
+    End Property
+
+    Public Shared Sub Main()
+        Dim prog As New Program()
+        prog.P = ""New value""
+    End Sub
+End Class
+"
+            Dim compilation = CreateCompilation(source, targetFramework:=TargetFramework.NetCoreApp, references:={Net451.MicrosoftVisualBasic}, options:=TestOptions.ReleaseExe, parseOptions:=TestOptions.RegularLatest)
+            CompileAndVerify(compilation, expectedOutput:="New value")
+            compilation.AssertTheseDiagnostics(
+<expected><![CDATA[
+BC42505: The CallerArgumentExpressionAttribute applied to parameter 'value' will have no effect. It is applied with an invalid parameter name.
+        Set(<CallerArgumentExpression("")> value As String)
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+]]></expected>)
+        End Sub
+
+        <ConditionalFact(GetType(CoreClrOnly))>
+        Public Sub TestSetter2()
+            Dim source As String = "
+Imports System
+Imports System.Runtime.CompilerServices
+
+Class Program
+    Public Property P As String
+        Get
+            Return ""Return getter""
+        End Get
+        Set(<CallerArgumentExpression(""value"")> value As String)
+            Console.WriteLine(value)
+        End Set
+    End Property
+
+    Public Shared Sub Main()
+        Dim prog As New Program()
+        prog.P = ""New value""
+    End Sub
+End Class
+"
+            Dim compilation = CreateCompilation(source, targetFramework:=TargetFramework.NetCoreApp, references:={Net451.MicrosoftVisualBasic}, options:=TestOptions.ReleaseExe, parseOptions:=TestOptions.RegularLatest)
+            CompileAndVerify(compilation, expectedOutput:="New value")
+            compilation.AssertTheseDiagnostics(
+<expected><![CDATA[
+BC42504: The CallerArgumentExpressionAttribute applied to parameter 'value' will have no effect because it's self-referential.
+        Set(<CallerArgumentExpression("value")> value As String)
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+]]></expected>)
+        End Sub
+
+        <ConditionalFact(GetType(CoreClrOnly))>
+        Public Sub TestSetter3()
+            Dim source As String = "
+Imports System
+Imports System.Runtime.CompilerServices
+
+Class Program
+    Public Property P As String
+        Get
+            Return ""Return getter""
+        End Get
+        Set(<CallerArgumentExpression("""")> Optional value As String = ""default"")
```

Comment:
>"""" [](http://example.com/codeflow?start=38&length=4)

Consider also testing a property with parameters and referring to one of the parameters here.